src/ListDate.elm

```diff
@@ -92,21 +101,103 @@ ensureSize size list =
 {-| Converts a date to a List of Int's. With the head being the year and the last
 the milliseconds.
 
-    dateToList date -- [ 2018, 5, 31,  0,  0,  0, 0 ]
-    dateToList date -- [ 2018, 5, 31, 15, 16,  0, 0 ]
-    dateToList date -- [ 2018, 5, 31, 15, 16, 20, 1000 ]
+    dateToList Time.utc (Posix 1527724800000) -- [ 2018, 5, 31,  0,  0,  0, 0 ]
+
+    dateToList Time.utc (Posix 1527779760000) -- [ 2018, 5, 31, 15, 16,  0, 0 ]
+
+    dateToList Time.utc (Posix 1527779780987) -- [ 2018, 5, 31, 15, 16, 20, 987 ]
 
 -}
-dateToList : Date -> List Int
-dateToList date =
+dateToList : Zone -> Posix -> List Int
+dateToList zone timestamp =
     -- Date
-    [ Date.year date
-    , DE.monthNumber date
-    , Date.day date
+    [ Time.toYear zone timestamp
+    , Time.toMonth zone timestamp |> monthToInt
+    , Time.toDay zone timestamp
 
     -- Time
-    , Date.hour date
-    , Date.minute date
-    , Date.second date
-    , Date.millisecond date
+    , Time.toHour zone timestamp
+    , Time.toMinute zone timestamp
+    , Time.toSecond zone timestamp
+    , Time.toMillis zone timestamp
     ]
+
+
+intToMonth : Int -> Month
+intToMonth month =
+    case month of
+        2 ->
+            Feb
+
+        3 ->
+            Mar
+
+        4 ->
+            Apr
+
+        5 ->
+            May
+
+        6 ->
+            Jun
+
+        7 ->
+            Jul
+
+        8 ->
+            Aug
+
+        9 ->
+            Sep
+
+        10 ->
+            Oct
+
+        11 ->
+            Nov
+
+        12 ->
+            Dec
+
+        _ ->
```

Comment:
Maybe a good idea to do a range check on this before and fail with an error if the month is out of range?

Looks like partsToPostix clamps values in range when presented with illegal input, which tbh is also a weird design choice... So it might not matter too much